images/origin/scripts/recycler.sh

```diff
@@ -4,40 +4,31 @@
 # this script is intended to be used in a pod that performs the scrub.
 # the container in the pod should succeed or fail based on the exit status of this script.
 
-set -o errexit
 set -o nounset
 set -o pipefail
 
-function recycle(){
+# first and only arg is the directory to scrub
+dir=${1:-}
 
-    # first and only arg is the directory to scrub
-    dir=$1
-
-    if [ -z "$dir" ]; then
-        echo "Usage:  scrub_directory some/path/to/scrub"
-        return 1
-    fi
+if [ -z "${dir}" ]; then
+    echo "Usage:  recycler.sh some/path/to/scrub"
+    exit 1
+fi
 
-    if [ ! -e $dir ]; then
-        echo "scrub directory $dir does not exist"
-        return 1
-    fi
+echo "Scrubbing $dir"
 
-    # remove everything but keep the directory for re-use as a volume
-    rm -rfv $dir/*
+# only delete the volume's contents
+# retain the directory for re-use by another pod.
+rm -rfv $dir/*
```

Comment:
I commented inline about this one, but the comment block above got collapsed.

rm doesn't support wildcard expansion, so $dir/* returned true (but removed nothing) and the script would wrongly pass.  

https://github.com/openshift/origin/pull/3657#discussion_r35032176